### DISCUSSION TITLE
[Info/CMake] How to extract the feature set of a CMake based port?

### DISCUSSION THREAD

**Discussion by Neumann-A** [microsoft/vcpkg#20709] [Q&A]
(should be put into the maintainers doc somewhere)

One major problem in CMake ports is to make sure all dependencies are found and that their are no optional dependencies which might screw CI up because sometimes those dependency can be available in CI and sometimes they are not. 

One way to find all dependencies is to do a search for all `find_package` and `find_library` calls within the source tree of that port. Another maybe less work intensive way is to use 
(https://cmake.org/cmake/help/latest/module/FeatureSummary.html)
```
include(FeatureSummary)
feature_summary(INCLUDE_QUIET_PACKAGES WHAT ALL)
```
at the end of the toplevel CMakeLists.txt and than check the end of the configure log. 
(this wont discover raw `find_library` calls not wrapped around a module but is already a big help)

**Comment by strega-nil** [comment]
Would you like to write up these docs?
